lib/tasks/after_tasks.rake

```diff
@@ -740,6 +740,35 @@ namespace :After do
     WorkIndexer.create_mapping
   end
 
+  desc "Fix tags with extra spaces"
+  task(fix_tags_with_extra_spaces: :environment) do
+    total_tags = Tag.count
+    total_batches = (total_tags + 999) / 1000
+    puts "Inspecting #{total_tags} tags in #{total_batches} batches"
+
+    report_string = ["Tag ID", "Old tag name", "New tag name"].to_csv
+    Tag.find_in_batches.with_index do |batch, index|
+      batch_number = index + 1
+      progress_msg = "Batch #{batch_number} of #{total_batches} complete"
+
+      batch.each do |tag|
+        next unless tag.name != tag.name.squish
+
+        old_tag_name = tag.name
+        new_tag_name = old_tag_name.gsub(/[[:space:]]/, "_")
+
+        new_tag_name << "_" while Tag.find_by(name: new_tag_name)
```

Comment:
Should we have a test for this situation, i.e. `Borked Tag_` exists when we are trying to fix `Borked Tag\u205F`, so the desired result is `Borked Tag__`?

Reply:
I think that's covered by this test case: https://github.com/otwcode/otwarchive/pull/4020/files#diff-aa2a58a681bbf3c1b89cffbc4736bcdd58f48271797eef3e7a129ab9a45692a3R168

I'm modifying `Borked\u00A0tag` when `Borked_tag` already exists and checking for `Borked_tag_` at the end.

Reply:
Oh, gotcha. I forgot we were replacing the internal spaces with underscores; I thought we were just converting them to regular spaces and adding underscores at the end. 👍